server/datastore/mysql/migrations/tables/20211109160905_PolicyWithProprietaryQuery.go

```diff
@@ -0,0 +1,70 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20211109160905, Down_20211109160905)
+}
+
+func Up_20211109160905(tx *sql.Tx) error {
+	if _, err := tx.Exec(`ALTER TABLE policies
+		ADD COLUMN name VARCHAR(255) NOT NULL,
+		ADD COLUMN query mediumtext NOT NULL,
+		ADD COLUMN description mediumtext NOT NULL,
+  		ADD COLUMN author_id int(10) unsigned DEFAULT NULL,
+		ADD team_id_x int(10) unsigned GENERATED ALWAYS AS (COALESCE(team_id, 0)) VIRTUAL NOT NULL,
+
+  		ADD KEY idx_policies_author_id (author_id),
+  		ADD KEY idx_policies_team_id (team_id),
+  		ADD CONSTRAINT policies_queries_ibfk_1 FOREIGN KEY (author_id) REFERENCES users (id) ON DELETE SET NULL
+	`); err != nil {
+		return errors.Wrap(err, "adding new columns to 'policies'")
+	}
+	// Remove duplicate global and team policy queries (references).
+	if _, err := tx.Exec(`
+        DELETE p1 FROM policies AS p1, policies AS p2
+		WHERE p1.ID < p2.ID
+		AND p1.query_id = p2.query_id AND p1.team_id <=> p2.team_id
```

Comment:
Not sure I follow this. Could you explain the logic behind this a bit more?

Reply:
Currently users are able to create "duplicate" policy queries (with same `query_id` as reference). This is deduplicating those. Deduplicating is needed to enforce uniqueness of `name+team_id` later on in the migration (see below).